lib/core/versions/latest/BatchWriter.ts

```diff
@@ -109,6 +102,43 @@ export default class BatchWriter implements IBatchWriter {
     await this.operationQueue.dequeue(queuedOperations.length);
   }
 
+  /**
+   * Create and write chunk file if needed.
+   * @returns CASE URI of the chunk file. `undefined` if there is no need to create and write the file.
+   */
+  private async createAndWriteChunkFileIfNeeded (
+    createOperations: CreateOperation[], recoverOperations: RecoverOperation[], updateOperations: UpdateOperation[]
+  ): Promise<string | undefined> {
+    const chunkFileBuffer = await ChunkFile.createBuffer(createOperations, recoverOperations, updateOperations);
+    if (chunkFileBuffer === undefined) {
+      return undefined;
+    }
+
+    const chunkFileUri = await this.cas.write(chunkFileBuffer);
+    console.info(LogColor.lightBlue(`Wrote chunk file ${LogColor.green(chunkFileUri)} to content addressable store.`));
+
+    return chunkFileUri;
+  }
+
+  /**
+   * Create and write provisional index file if needed.
+   * @returns CASE URI of the chunk file. `undefined` if there is no need to create and write the file.
+   */
+  private async createAndWriteProvisionalIndexFileIfNeeded (
+    chunkFileUri: string | undefined, provisionalProofFileHash: string | undefined, updateOperations: UpdateOperation[]
+  ): Promise<string | undefined> {
+    // If `chunkFileHash` is `undefined` it means there are only deactivates, and a batch with only deactivates does not reference a provisional index file.
```

Comment:
```suggestion
    // If `chunkFileUri` is `undefined` it means there are only deactivates, and a batch with only deactivates does not reference a provisional index file.
```

Reply:
Thanks, replaced all occurrences in code base.